docs/ops/matrix/Einsum_7.md

```diff
@@ -0,0 +1,248 @@
+
+*Einsum* operation on multiple inputs can be treated as several consecutive *Einsum* operations. In the first step, *Einsum* applies the first two inputs. 
+In the second step, it operates on the result of the first step and the third input and so forth.
+*Einsum* operates on two operands similar to element-wise multiplication by all pairs of batches from both operands. The batch dimensions are defined
+with labels that enter only one subscript from the corresponding two subscripts.
+For example, the intermediate result after the first step for *Einsum* with three inputs (of shapes `[2, 5]`, `[5, 3, 6]` and `[5, 3]`)
+and `equation` equal to `ab,bcd,bc->ca` will be a tensor of shape `[2, 5, 3, 6]` with a subscript `abcd`
+where batch dimensions for the first input and the second input are represented with label sequences `a` and `cd`.
+The next step performs the same logic on input tensors (of shapes `[2, 5, 3, 6]` and `[5, 3]`) with subscripts `abcd` and `bc` and
+outputs a tensor of shape `[2, 5, 3, 6]` with a subscript `abcd`.
+Lastly, the output subscript defines order of output dimensions and which dimensions are sum-reduced.
+Dimensions corresponding to absent labels in the output subscript are sum-reduced. The final result for the considered example is of shape equal to `[3,2]`
+where dimensions with labels `b` and `d` are reduced and the transpose is applied to get output layout `ca`.
+
+**NOTE**: *Einsum* operation can perform on the single operand and in this case the operation can transpose the input and reduce its dimensions.
+
+**NOTE**: Input ranks must be equal to a length of corresponding subscripts and dimensions to which the same labels in input subscripts 
+correspond must be equal in size.
+
+**NOTE**: A labels can repeat in the same input subscript, for example, `equation` equal to `aac,abd,ddde`. In this case the corresponding dimensions
+must match in size and the operand the operand will be replaced by its diagonal along these dimensions.
+For example, *Einsum* operation on the single 3D tensor of shape `[2, 4, 5, 4]` with `equation` equal to `ijkj->ij`.
+
+**NOTE**: The specification considers the primitive algorithm for *Einsum* operation for better understanding of the operation
+and does not recommend it for implementation.
+
+**NOTE**: The described algorithm can be improved by immediate dimension sum-reduction in the intermediate results if the corresponding labels absent in
+the input subscripts of subsequent inputs and the output subscript. It can significantly boosts performance and reduces memory costs.
+In the considered example, after the first step we can reduce a dimension corresponding to a label `d`.
+
+The output shape can be easily computed using the output subscript. It needs to concatenate sizes of dimensions to which labels in output subscript
+correspond in the specified order.
```

Comment:
It needs to concatenate sizes of dimensions to which labels in output subscript
correspond in the specified order. -> don't like the wording, but hard to come out with a better sentence, could you rephrase it?

Reply:
I can rephrase these two sentences as follows:
"The output shape is computed by concatenation of dimension sizes to which labels in the output subscript correspond in the specified order."

Reply:
Done.